### PR TITLE
Update MacOS, Windows, and Android SDK version

### PR DESCRIPTION
**Pull request by o01eg**
Don't use Python framework

### CODE REVIEW

**Review by Vezzra — APPROVED** [review]
Works! 👍

File: CMakeLists.txt [hunk 1]
```diff
@@ -102,11 +102,11 @@ else()
 endif()
 
 if(APPLE)
-    set(FreeOrionSDK_VERSION 13)
+    set(FreeOrionSDK_VERSION 14)
     set(FreeOrionSDK_FILE "FreeOrionSDK_${FreeOrionSDK_VERSION}_Clang-MacOSX-10.9-x86_64.tar.bz2")
     if(NOT EXISTS "${CMAKE_BINARY_DIR}/${FreeOrionSDK_FILE}")
     file(DOWNLOAD
-            "https://github.com/freeorion/freeorion-sdk/releases/download/v${FreeOrionSDK_VERSION}/${FreeOrionSDK_FILE}"
```
Comment (Cjkjvfnby) [inline CMakeLists.txt:109]: This points to the fork

File: CMakeLists.txt [hunk 2]
```diff
@@ -102,11 +102,11 @@ else()
...
+            "https://github.com/o01eg/freeorion-sdk/releases/download/v${FreeOrionSDK_VERSION}/${FreeOrionSDK_FILE}"
```
Comment (Cjkjvfnby) [inline CMakeLists.txt]: @o01eg , this line points to your fork

Comment (o01eg) [reply CMakeLists.txt]: Yes, it to test if MacOS works with upcoming SDK release. It will be changed back after the release.

**Review by Cjkjvfnby — APPROVED** [review]

**Comment by o01eg** [comment]
I've updated Windows CMake build too. Somehow windows-2019 doesn't auto-detect Python from SDK while windows-2022 does, so I set Python path to cmake arguments.

**Comment by o01eg** [comment]
Updated it to use SDK with boost 1.82.0

**Comment by o01eg** [comment]
Updated SDK to include https://github.com/freeorion/freeorion-sdk/pull/116

**Comment by o01eg** [comment]
https://github.com/freeorion/freeorion/actions/runs/4776351620/jobs/8491325644?pr=4482

**Comment by o01eg** [comment]
Switched it to SDK release 14 and squashed.

**Comment by o01eg** [comment]
Superseded by #4502

**Pull request closed without merging**